CONTRIBUTING.md

```diff
@@ -136,3 +136,39 @@ Once you've done that, you can run the `citus_indent` command to recursively che
 correct the style of any source files in the current directory. You can also run `make
 reindent` from within the Citus repo to correct the style of all source files in the
 repository.
+
+### Making SQL changes
+
+Sometimes you need to make change to the SQL that the citus extension runs upon
+creations. The way this is done is by changing the last file in
+`src/backend/distributed/sql`, or creating it if the last file is from a
+published release. If you needed to create a new file, also change the
+`default_version` field in `src/backend/distributed/citus.control` to match your
+new version. All the files in this directory are run in order based on
+their name. See [this page in the Postgres
+docs](https://www.postgresql.org/docs/current/extend-extensions.html) for more
+information on how Postgres runs these files.
+
+#### Changing or creating functions
+
+If you need to change any functions defined by Citus. You should check inside
+`src/backend/distributed/sql/udfs` to see if there is already a directory for
+this function, if not create one. Then change or create the file called
+`latest.sql` in that directory to match how it should create the function. This
+should be including any DROP (IF EXISTS), COMMENT and REVOKE statements for this
+function.
+
+Then copy the `latest.sql` file to `{version}.sql`, where `{version}` is the
+version for which this sql change is, e.g. `{9.0-1.sql}`. Now that you've
+created this stable snapshot of the function definition for your version you
+should use it in your actual sql file, .e.g.
+`src/backend/distributed/sql/citus--8.3-1--9.0-1.sql`. You do this by using C
```

Comment:
I think I need a clarification on this. When I look at the mentioned [file](https://github.com/citusdata/citus/blob/master/src/backend/distributed/sql/citus--8.3-1--9.0-1.sql), I see `worker_create_or_replace_object ()`. 

For example, why is that not included via a header and explicitly added? (Or may other function (re)-definitions on the file)

Reply:
Mostly because I pushed back on this a bit. I think there are some risks to liberally separating UDFs into separate SQL files such as bugs around permissions, search paths, and ambiguity around how to declare types that are used in a function argument. We should form a more comprehensive policy around how we do SQL scripts.

On the other hand, the lack of history tracking for PL/pgSQL has been very annoying and this release introduces lengthy upgrade scripts. It seems better not to wait with those since they are likely to change in the near future.